### ISSUE TITLE
[EGL] Unable to acquire context: EGL_BAD_ALLOC: EGL failed to allocate resources for the requested operation.

### ISSUE THREAD

**Issue by martin3361** [googleads/googleads-mobile-unity#711]
Strangely, I started getting this error some days ago, without changing anything - Android version stayed the same, Admob Unity plugin stayed the same, Unity version stayed the same. But all of a sudden:
`[EGL] Unable to acquire context: EGL_BAD_ALLOC: EGL failed to allocate resources for the requested operation.`
This happens when a video interstitial is loaded, and my application in the background is obviously frozen, because I have made a workaround of Admob ads not pausing my application, which is [posted here](https://github.com/googleads/googleads-mobile-unity/issues/546). Then my application stops pinging the server and whatever else it is doing in the background.
Then time comes to close the ad, and when getting back to my activity, the above error is printed.

What amuses me most about this problem is that it started happening without me changing anything.

I'm using AdMob Unity plugin 3.7.0.

**Comment by martin3361** [comment]
I am attaching the logs I obtained with `adb`. The logs start with the exact moment the ad was triggered and end a bit after it was closed (and ad should return to app context).
This specific snippet contains the error:

> 07-05 12:14:19.216   743  1502 V WindowManager: Relayout window turning screen on: Window{93c885d u0 com.classicjim.bomberz/com.classicjim.bomberz.CustomUnityPlayerActivity}
07-05 12:14:19.217   239   239 I BufferQueue: [unnamed-239-819](this:0x7cedb37800,id:819,api:0,p:-1,c:-1) BufferQueue core=(239:/system/bin/surfaceflinger)
07-05 12:14:19.217   239   239 I BufferQueueConsumer: [unnamed-239-819](this:0x7cedb37800,id:819,api:0,p:-1,c:239) connect(C): consumer=(239:/system/bin/surfaceflinger) controlledByApp=false
07-05 12:14:19.217   239   239 I BufferQueueConsumer: [unnamed-239-773](this:0x7cedb37800,id:819,api:0,p:-1,c:239) setConsumerName: unnamed-239-773
07-05 12:14:19.218   239   239 I BufferQueueConsumer: [com.classicjim.bomberz/com.classicjim.bomberz.CustomUnityPlayerActivity](this:0x7cedb37800,id:819,api:0,p:-1,c:239) setConsumerName: com.classicjim.bomberz/com.classicjim.bomberz.CustomUnityPlayerActivity
07-05 12:14:19.218   239   239 I BufferQueueProducer: [](this:0x7cedb37800,id:819,api:0,p:-1,c:239) setMaxDequeuedBufferCount: maxDequeuedBuffers = 2
07-05 12:14:19.218   239   239 I BufferQueueConsumer: [com.classicjim.bomberz/com.classicjim.bomberz.CustomUnityPlayerActivity](this:0x7cedb37800,id:819,api:0,p:-1,c:239) setDefaultBufferSize: width=1024 height=600
07-05 12:14:19.218   239  2486 E BufferQueueProducer: [SurfaceView - com.classicjim.bomberz/com.classicjim.bomberz.CustomUnityPlayerActivity](this:0x7cedbe0800,id:812,api:1,p:23141,c:-1) dequeueBuffer: BufferQueue has been abandoned
07-05 12:14:19.220 23141 23157 E IMGSRV  : :0: DequeueBufferWrapper: Failed to de-queue buffer (-19)
07-05 12:14:19.220 23141 23157 E IMGSRV  : :0: KEGLGetDrawableParameters: Native window is invalid
07-05 12:14:19.220 23141 23157 E IMGSRV  : :0: GLES3MakeCurrentGC: Invalid write drawable - what do we do?
07-05 12:14:19.220 23141 23157 E IMGSRV  : :0: IMGeglMakeCurrent: unexpected error code
07-05 12:14:19.220 23141 23157 E libEGL  : eglMakeCurrent:856 error 3003 (EGL_BAD_ALLOC)
07-05 12:14:19.221 23141 23157 E Unity   : [EGL] Unable to acquire context: EGL_BAD_ALLOC: EGL failed to allocate resources for the requested operation.
07-05 12:14:19.221 23141 23157 E Unity   :  
07-05 12:14:19.221 23141 23157 E Unity   : (Filename: /Users/builduser/buildslave/unity/build/Runtime/GfxDevice/egl/WindowContextEGL.cpp Line: 267)
07-05 12:14:19.221 23141 23157 E Unity   : 
07-05 12:14:19.221 23141 23157 D Surface : Surface::disconnect(this=0xdc352500,api=1)
07-05 12:14:19.221 23141 23157 D GraphicBuffer: unregister, handle(0xc6d83180) (w:600 h:1024 s:608 f:0x1 u:0x000b00)
07-05 12:14:19.221 23141 23157 D IMGSRV  : GrallocUnregisterBuffer:1969: bFree=0 ID=197379 ref=1
07-05 12:14:19.221 23141 23157 D GraphicBuffer: unregister, handle(0xb6f8c400) (w:600 h:1024 s:608 f:0x1 u:0x000b00)
07-05 12:14:19.221 23141 23157 D IMGSRV  : GrallocUnregisterBuffer:1969: bFree=0 ID=197390 ref=1
07-05 12:14:19.222 23141 23157 D GraphicBuffer: unregister, handle(0xb6f8c480) (w:600 h:1024 s:608 f:0x1 u:0x000b00)
07-05 12:14:19.222   743  1502 V WindowManager: Changing focus from null to Window{93c885d u0 com.classicjim.bomberz/com.classicjim.bomberz.CustomUnityPlayerActivity} 

[logs.txt](https://github.com/googleads/googleads-mobile-unity/files/2165993/logs.txt)
